src/main/java/bio/terra/cloudfunctions/proto/ProtoApp.java

```diff
@@ -0,0 +1,62 @@
+package bio.terra.cloudfunctions.proto;
```

Comment:
I think the best approach is to use a raw function.  How about something like: 
```
public void accept(String s, Context context) throws Exception {
    logger.info(String.format("Here's the message %s", s));
    StorageObjectData storageObjectData =
        GsonConverter.convertFromClass(s, StorageObjectData.class);
    logger.info(String.format("Here's the serialized object %s", storageObjectData));
    new ProtoApp(convertMessage(storageObjectData)).process();
  }
````

Reply:
In considering `RawBackgroundFunction` vs `CloudEventsFunction`.

My preference would be the more basic `CloudEventsFunction` that implements `public void accept(CloudEvent event) throws Exception`. For example, consider the following scenario.

What about if Google Storage version changes say from `v1 -> v2`) and nobody informs us? The following code will likely throw an exception:
`StorageObjectData storageObjectData =
        GsonConverter.convertFromClass(s, StorageObjectData.class);`

We can more easily anticipate such changes by using the `CloudEvent` 'specVersion`. I am unsure if the `Context` object in `RawBackgroundFunction.accept` method can capture this type of information.

Reply:
If Google changes their JSON payload and doesn't tell anyone beforehand or doesn't preserve the existing response type under versioning, I don't think a cloud events function is any more insulated from that change than a raw function would be, right?  Am I missing something?